builder/struct.go

```diff
@@ -40,6 +40,33 @@ func (*Struct) Build(gen Generator, ctx *MethodContext, sourceID *xtype.JenID, s
 		}
 
 		targetFieldType := xtype.TypeOf(targetField.Type())
+
+		if mapFnName, ok := ctx.ExtendMapping[targetField.Name()]; ok {
+			if sourceMatch, err := source.StructField(targetField.Name(), ctx.MatchIgnoreCase, ctx.IgnoredFields); err == nil {
```

Comment:
As described in https://github.com/jmattheis/goverter/issues/12#issuecomment-1010274080 let's start with only implementing the method without parameters.

> The first step would be to add goverter:mapExtend with support for only defining a method without arguments and one return value, which must be the same as the return type of the parameter that is defined inside goverter:mapExtend. The optional passing of the source value as first parameter should be done later.

Reply:
In my use case , pass single parameter into mapExtend function is needed,
currently, struct type or source type  as first parameter  are fully supported and tested

```golang
  func PluckAgeInt(source Input) int {
      return source.Age
  }

  func PlusScoreInt(source int) int {
      return source + 1
  }
```

Reply:
Yes, but it is much easier to review if there are fewer features in one pull request. The use-case that was originally proposed in the ticket is not supported, and I don't think the current structure allows this to be cleanly implemented. (mapping extend method without parameter)

```go
// goverter:converter
type Converter interface {
    // goverter:mapExtend Age IntOne
    Convert(source Input) Output
}

type Input struct {
    Name string
}
type Output struct {
    Name string
    Age int
}
func IntOne() int {
    return 1
}
```

The BuildExtend method has code for returning errors and passing the converter instance to the extend method, this all is not tested.

Reply:
add  mapping extend method without parameter in commit https://github.com/jmattheis/goverter/pull/25/commits/e6688d753033217015307a6a5bf955833ce09cf1

Reply:
To make it simple,  I just remove support for returning errors and passing the converter instance to the extend method currently in commit https://github.com/jmattheis/goverter/pull/25/commits/4962136c2cb01ba62cabc19acc1242ba85a83ba6